keras_cv/models/convnext.py

```diff
@@ -0,0 +1,539 @@
+    """
+    if name is None:
+        name = "prestem" + str(backend.get_uid("prestem"))
+
+    def apply(inputs):
+        x = inputs
+
+        x = layers.Conv2D(
+            filters=projection_dim,
+            kernel_size=7,
+            padding="same",
+            groups=projection_dim,
+            name=name + "_depthwise_conv",
+        )(x)
+        x = layers.LayerNormalization(epsilon=1e-6, name=name + "_layernorm")(x)
+        x = layers.Dense(4 * projection_dim, name=name + "_pointwise_conv_1")(x)
+        x = layers.Activation("gelu", name=name + "_gelu")(x)
+        x = layers.Dense(projection_dim, name=name + "_pointwise_conv_2")(x)
+
+        if layer_scale_init_value is not None:
+            x = LayerScale(
+                layer_scale_init_value,
+                projection_dim,
+                name=name + "_layer_scale",
+            )(x)
+        if drop_path_rate:
+            layer = StochasticDepth(drop_path_rate, name=name + "_stochastic_depth")
+            return layer([inputs, x])
+        else:
+            layer = layers.Activation("linear", name=name + "_identity")
+            return inputs + layer(x)
+
+    return apply
+
+
+def Head(num_classes, activation="softmax", name=None):
+    """Implementation of classification head of RegNet.
+
+    Args:
+      num_classes: number of classes for Dense layer
+      activation: activation function for Dense layer
+      name: name prefix
+
+    Returns:
+      Classification head function.
+    """
+    if name is None:
+        name = str(backend.get_uid("head"))
+
+    def apply(x):
+        x = layers.GlobalAveragePooling2D(name=name + "_head_gap")(x)
+        x = layers.LayerNormalization(epsilon=1e-6, name=name + "_head_layernorm")(x)
+        x = layers.Dense(num_classes, activation=activation, name=name + "_head_dense")(
+            x
+        )
+        return x
+
+    return apply
+
+
+def ConvNeXt(
+    include_rescaling,
+    include_top,
+    depths,
+    projection_dims,
+    drop_path_rate=0.0,
+    layer_scale_init_value=1e-6,
+    weights=None,
+    input_shape=(None, None, 3),
+    input_tensor=None,
+    pooling=None,
+    classes=None,
+    classifier_activation="softmax",
+    name="convnext",
+):
+    """Instantiates ConvNeXt architecture given specific configuration.
+
+    Args:
+      include_rescaling: whether or not to Rescale the inputs. If set to True,
+        inputs will be passed through a `Rescaling(1/255.0)` layer.
+      include_top: Boolean denoting whether to include classification head to
+        the model.
+      depths: An iterable containing depths for each individual stages.
+      projection_dims: An iterable containing output number of channels of
+      each individual stages.
+      drop_path_rate: Stochastic depth probability. If 0.0, then stochastic
+        depth won't be used.
+      layer_scale_init_value: Layer scale coefficient. If 0.0, layer scaling
+        won't be used.
+      weights:
```

Comment:
Same as https://github.com/keras-team/keras-cv/pull/678/files#r947004467.

Reply:
(Same -- let's use the string "one of `None` (random initialization), or a pretrained weight file path")

Reply:
Can you suggest a text?

Reply:
Yes, sorry I wasn't clear.

Let's use "one of None (random initialization), or a pretrained weight file path", which is the same as the docstrings from the other models which don't yet support pre-trained weights.